tests/magnetostatics/test_polyhedral_prism.py

```diff
@@ -382,3 +382,89 @@ def test_triangle(self, plane):
         cm = ax.contourf(args_new[i], args_new[j], args_new[k], zdir=plane, offset=0)
         f.colorbar(cm)
         plt.show()
+
+
+class TestCombinedShapes:
+    @classmethod
+    def setup_class(cls):
+        current = 1e6
+        coords = Coordinates({"x": [-1, -1, 1, 1], "z": [1, -1, -1, 1]})
+        cls.square = PolyhedralPrismCurrentSource(
+            [0, 0, 0],
+            [10, 0, 0],
+            [0, 1, 0],
+            [0, 0, 1],
+            coords,
+            10,
+            10,
+            current,
+        )
+        coords = Coordinates({"x": [-1, -1, 1], "z": [1, -1, -1]})
+        cls.triangle1 = PolyhedralPrismCurrentSource(
+            [0, 0, 0],
+            [10, 0, 0],
+            [0, 1, 0],
+            [0, 0, 1],
+            coords,
+            10,
+            10,
+            current / 2,
+        )
+        coords = Coordinates({"x": [-1, 1, 1], "z": [1, -1, 1]})
+        cls.triangle2 = PolyhedralPrismCurrentSource(
+            [0, 0, 0],
+            [10, 0, 0],
+            [0, 1, 0],
+            [0, 0, 1],
+            coords,
+            10,
+            10,
+            current / 2,
+        )
+
+    @pytest.mark.parametrize("plane", ["x", "y", "z"])
+    def test_plot(self, plane):
+        n = 50
+        x1, x2 = np.linspace(-5, 5, n), np.linspace(-5, 5, n)
+        xx1, xx2 = np.meshgrid(x1, x2)
+        xx3 = np.zeros_like(xx1)
+
+        if plane == "x":
+            xx, yy, zz = xx3, xx1, xx2
+            i, j, k = 3, 1, 2
+        elif plane == "y":
+            xx, yy, zz = xx1, xx3, xx2
+            i, j, k = 0, 3, 2
+        elif plane == "z":
+            xx, yy, zz = xx1, xx2, xx3
+            i, j, k = 0, 1, 3
```

Comment:
could you pull this out into a function in this file, it seems like its used loads of times